### PR TITLE
zoom-us: 2.2.128100.0627 -> 2.2.128200.0702

### LINKED ISSUE #41203: Zoom black video on NixOS
## Issue description
Currently Zoom on master launches, but the camera video is black. This happens to be with one of updated SYSTEM packages.

So far, I've figured the following:

1. This is SYSTEM issue. I've found that with following Nixpkgs revisions it works:
```
  nixos-2018-04-04 = mkNixpkgs {
    rev = "ea145b68a01";
  };

  nixos-pre-libvnd-change = mkNixpkgs {
    rev = "01b254ec50299f1fe1d3e9a4eb9a002f6fe0c84d";
    patches = [
      { meta.description = "ostree-patch-new-version";
        url = "https://github.com/NixOS/nixpkgs/commit/546696e367fe94159aa7765313d93caffd11fb37.patch";
        sha256 = "01jmdfzzcsidmah269m51020g2waf7j4dbnwr7chif8cci8096qy";
        name = "ostree-patch.diff";
      }
      { meta.description = "fix-build-webkitgtk";
        url = "https://github.com/NixOS/nixpkgs/pull/38375/commits/50585295724de4d971dec1e4258daf047ce7da54.patch";
        sha256 = "1mjf0ydrfl6js1flv4hgg1i0gcamxwpd0yspgd9ai0aalnjqh6m4";
        name = "webkitgtk.diff";
      }
    ];
  };
```
2. Zoom from master also works on that revision (camera video works)
3. I haven't yet bisected this, but with following revision camera video doesn't work
```
  nixos-2018-05-21 = mkNixpkgs {
    rev = "5da85431fb1";
  };
```
4. It may be of interest - I'm using Gnome Shell as DM.
5. Running nix-diff on my system closures revealed the following changes:
```
    - dbus-1.10.24
    - e2fsprogs-1.44.0
    - fuse-3.2.1
    - linux-4.15.12
    - sudo-1.8.22
    + dbus-1.12.6
    + e2fsprogs-1.44.1
    + fuse-3.2.3
    + linux-4.16.10
    + perl-Net-DBus-1.1.0
    + perl-XML-Parser-2.44
    + perl-XML-Twig-3.52
    + sudo-1.8.23
```
I assume that this happens to be something with DBus or kernel then (Zoom has own DBus in Qt bundle)

### Steps to reproduce
Install Zoom and launch new video conference. Screen is black.

This probably isn't dupe of https://github.com/NixOS/nixpkgs/issues/40776, but perhpas related

### PR DESCRIPTION
**Pull request by tadfisher**
###### Motivation for this change

Fix #41203. The `zoom` binary loads `./libturbojpeg.so` from its CWD, so link the lib and change the CWD in the wrapper.

Also adds `libpulseaudio` and `pciutils` to runtime paths after noticing these are used.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 1]
```diff
@@ -52,8 +54,19 @@ in stdenv.mkDerivation {
...
+      ln -s $(readlink -e "${libjpeg_turbo.out}/lib/libturbojpeg.so") $packagePath/libturbojpeg.so
+
+      cat >$out/bin/zoom-us <<EOF
+      #!${stdenv.shell} -e
+      export QSG_INFO=1
+      export PATH="${stdenv.lib.makeBinPath [ pciutils procps ]}"${"\\\${PATH:+':'}"}"\$PATH"
+      export LD_LIBRARY_PATH="${stdenv.lib.makeLibraryPath [ libpulseaudio ]}"${"\\\${LD_LIBRARY_PATH:+':'}"}"\$LD_LIBRARY_PATH"
+      cd $packagePath
+      exec ./zoom "\$@"
+      EOF
```
Comment (tadfisher) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: If there's a way to do this with `makeWrapper`, please let me know.

Comment (tadfisher) [reply pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: (or if it's possible to patch the binary somehow)

**Comment by tadfisher** [comment]
cc maintainers: @danbst

**Comment by danbst** [comment]
I can't start zoom now
```
zoom started.
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```
I had same problem in past and fixed it by supplying qt.conf:

https://github.com/tadfisher/nixpkgs/blob/16463d728b81f159e3b5f4eaae241841b227d86f/pkgs/applications/networking/instant-messengers/zoom-us/default.nix#L83-L86

Maybe this is because I use Gnome Shell as desktop?

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 2]
```diff
@@ -52,8 +64,13 @@ in stdenv.mkDerivation {
 
       cp -ar ${files} $packagePath
 
+      # TODO Patch this somehow; tries to dlopen './libturbojpeg.so' from cwd
+      ln -s $(readlink -e "${libjpeg_turbo.out}/lib/libturbojpeg.so") $packagePath/libturbojpeg.so
+
       makeWrapper $packagePath/zoom $out/bin/zoom-us \
-        --prefix PATH : "${procps}/bin"
+        --prefix PATH : "${makeBinPath [ pciutils procps ]}" \
```
Comment (danbst) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: Why is `pciutils` added to runtime PATH?

File: pkgs/applications/networking/instant-messengers/zoom-us/default.nix [hunk 3]
```diff
@@ -24,11 +30,17 @@ in stdenv.mkDerivation {
   buildInputs = [
     dbus glib libGL libX11 libXfixes libuuid libxcb qtbase qtdeclarative
     qtlocation qtquickcontrols2 qtscript qtwebchannel qtwebengine
+    libjpeg_turbo pciutils procps
```
Comment (danbst) [inline pkgs/applications/networking/instant-messengers/zoom-us/default.nix]: I think `pciutils` and `procps` shouldn't be mentioned here. We specify them as build inputs as arguments in package args, but those shouldn't be discovered later by build script - we refer to those explicitly.

IMO `buildInputs` are formed from packages that are used implicitly by build script. pciutils and procps are noop here.

**Comment by danbst** [comment]
Quoting myself:

> I had same problem in past and fixed it by supplying qt.conf:

I've checked that `qt.conf` doesn't fix the issue. Probably those (`libxcb.so`, for example) should be present on `LD_LIBRARY_PATH`

**Pull request merged**